### ISSUE TITLE
First run should prompt user to set timezone

### ISSUE THREAD

**Issue by beroset** [AsteroidOS/asteroid-launcher#172]
# The problem
The first time the watch starts shows a short tutorial and then guides the user through setting the language, time and date.  However, as @robthebold pointed out in a recent Matrix chat, the result is that the time is set in UTC rather than local time.  Since having the timezone correct is important for travelers and anyone who synchronizes the calendar, the first run should prompt for the timezone after the language is set and before the time and date are set.

# Possible fixes
Today, language, time and date settings are in https://github.com/AsteroidOS/asteroid-launcher/blob/master/src/qml/firstrun/FirstRunConfig.qml and largely duplicate those settings from `asteroid-settings`.  We could:

1. continue that model by duplicating and adapting the more complex timezone settings into asteroid-launcher
    **Advantages:**
      a. straightforward and consistent with other code in `FirstRunConfig.qml`
      b. does not require changes to any other code
    **Disadvantages:**
      a. large duplicated code which has to be separately maintained
      b. changes in structure mean that code can't be copied from `asteroid-settings` without modification
2. copy the relevant settings pages and then run them using a state machine in `FirstRunConfig.qml`
    **Advantages:**
      a. copying the entire pages would make it easier to maintain over time
      b. may make the structure and state machine more visible to programmers
      c. keeps the settings separate from the driving logic
    **Disadvantages:**
      a. large duplicated code which has to be separately maintained
      b. more of `asteroid-settings` layer infrastructure would also probably have to be copied
3. adapt asteroid-settings to be directly driven by `asteroid-launcher`
    **Advantages:**
      a. avoids duplicating code
      b. provides a general mechanism which might be useful elsewhere
    **Disadvantages:**
      a. more complex than the current cut-and-paste solution
      b. requires changes to two projects: both `asteroid-launcher` and `asteroid-settings`

**Comment by eLtMosen** [comment]
Do i understand correct that, in layman terms, the goal of #3 is to call the existing settings pages for language, timezone, date and time from within the asteroid-launcher during the initial setup?

**Comment by beroset** [comment]
> Do i understand correct that, in layman terms, the goal of #3 is to call the existing settings pages for language, timezone, date and time from within the asteroid-launcher during the initial setup?

Yes, that's it exactly.  I hope to prototype this approach soon just to try it out.

**Comment by beroset** [comment]
Actually, the reason timezones were introduced was to correctly support our calendar application.  Imported calendars require accurate timezone to function.

**Comment by moWerk** [comment]
Yes, indeed. I think i will pick this up later since its interesting enough to explore. Thanks for the swift review and merge of Option 1!
